lib/trino-orc/src/main/java/io/trino/orc/TupleDomainOrcPredicate.java

```diff
@@ -142,10 +145,10 @@ private boolean columnOverlaps(Domain predicateDomain, long numberOfRows, Column
         return discreteValues.get().stream().anyMatch(value -> checkInBloomFilter(bloomFilter, value, stripeDomain.getType()));
     }
 
-    private static Optional<Collection<Object>> extractDiscreteValues(ValueSet valueSet)
+    private Optional<Collection<Object>> extractDiscreteValues(ValueSet valueSet)
     {
         if (!valueSet.isDiscreteSet()) {
-            return Optional.empty();
+            return tryExpandRanges(valueSet, expandDiscreteValuesLimit);
```

Comment:
A compacted Domain would be encountered here mainly due to the low default value of `hive.domain-compaction-threshold` (100).
We've made many improvements around that class since that threshold was decided. So can we just look at increasing that threshold to 1000 instead ? Parquet reader can also potentially benefit from a more granular Domain when it tries to perform row group pruning using column indexes or dictionaries.
Also, if we really have to do this, can we consider a different approach to this API where we expose a way to iterate over the ranges instead of constructing a List/Set out of it ? (maybe similar to ValuesProcessor but for discrete values contained within the ranges). That way we can bail out of bloom filter lookups on the first positive match and there is less need to put an arbitrary bound like `orc_bloom_filters_expand_discrete_values_limit` which looks very difficult for end user to understand and tune.

Reply:
It also may happen that the original query predicate is a range (e.g. `col BETWEEN 1 AND 1000000`) so we don't want to expand it into a huge list of values - see https://github.com/trinodb/trino/pull/3316#issuecomment-956193739.

>  there is less need to put an arbitrary bound like `orc_bloom_filters_expand_discrete_values_limit` which looks very difficult for end user to understand and tune.

I agree :)

Reply:
Ok, I see now this can potentially benefit bounded range predicates, I was thinking only about equality and IN predicates earlier. This seems more useful in context of hive bucket filter than orc bloom filter as we're less likely to encounter integral data type with bloom filter (orc bloom filter itself seems rarely used feature).